app/ui/legacy/src/main/java/com/fsck/k9/activity/LauncherShortcuts.kt

```diff
@@ -0,0 +1,56 @@
+package com.fsck.k9.activity
+
+import android.content.Intent
+import android.os.Bundle
+import com.fsck.k9.BaseAccount
+import com.fsck.k9.activity.MessageList.Companion.shortcutIntent
+import com.fsck.k9.activity.MessageList.Companion.shortcutIntentForAccount
+import com.fsck.k9.search.SearchAccount
+import com.fsck.k9.ui.R
+
+class LauncherShortcuts : AccountList() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setTitle(R.string.shortcuts_title)
+
+        // finish() immediately if we aren't supposed to be here
+        if (Intent.ACTION_CREATE_SHORTCUT != intent.action) {
+            finish()
+        }
+    }
+
+    override fun onAccountSelected(account: BaseAccount) {
+        val shortcutIntent: Intent
+        if (account is SearchAccount) {
+            shortcutIntent = shortcutIntent(this, account.id)
+        } else {
+            shortcutIntent = shortcutIntentForAccount(this, account.uuid)
+        }
+
+        val displayName = account.name ?: account.email
+        val iconResId = R.drawable.ic_launcher
+        val iconResource = Intent.ShortcutIconResource.fromContext(this, iconResId)
+
+        setResult(
+            RESULT_OK,
+            createResultIntent(
+                shortcutIntent,
+                displayName,
+                iconResource,
+            ),
+        )
+        finish()
+    }
+
+    private fun createResultIntent(
+        shortcutIntent: Intent,
+        displayName: String,
+        iconResource: Intent.ShortcutIconResource,
+    ): Intent {
+        return Intent(Intent.ACTION_VIEW).apply {
```

Comment:
The result intent is not [documented](https://developer.android.com/reference/android/content/Intent#ACTION_CREATE_SHORTCUT) to contain an action.

```suggestion
        return Intent().apply {
```